### ISSUE TITLE
Generate Diff Between Save Files

### ISSUE THREAD

**Issue by nelsam** [pgmodeler/pgmodeler#1539]
**Feature description**
We would like to generate a diff between one `.dbm` file and another `.dbm` file, rather than a running database and a `.dbm` file.

**Additional info**
We are using [`migrate`](https://github.com/golang-migrate/migrate) to migrate between versions of our database model.  The diff tool is _very close_ to what we need to support migrations.

However, not all of our tables are part of the same model file.  This is partly for future compatibility - a separation of concerns.  This means that if we spin up a database using our migration files, the import does not match the layout that pgmodeler expects when generating a diff - there are tables not part of the `.dbm` file that it gets confused about and tries to drop.

We recognize that we could probably pull this off by using pgmodeler's CLI to apply the `.dbm` file to an empty database and generate a diff from that, but (a) I don't like the proposition that we'd have to _trust_ that database to match the save file (this will be run on developer machines, and some developer some day _will_ manage to generate a diff against the wrong database - our tooling should not require devs to double-check a running database to generate a diff), and (b) we're still experiencing some import failures related  to timescaledb that make it hard for us to trust a diff based on an import of a running database.

**Comment by nelsam** [comment]
> As I could understand you can tweak this behavior by performing [partial diffs](https://pgmodeler.io/blog/2020/7/10/im-glad-to-announce-pgmodeler-093-beta). This way, pgModeler will operate only over the set of objects that you specify. Anyway, if you provide more details I can help you more assertively.

Yeah, pgmodeler provides plenty of ways to do what we need using imports, and we've got plans to _start_ there.  This feature isn't a _hard_ blocker to do what we need to do, but we still consider these to be workarounds.

> I understand... This is a requirement of your development environment/process but, just to be clear, the diff process is a reliable process. It's not perfect, but in almost all cases it generates perfect code to synchronize a model and a running DB (or two databases if you prefer).

Yeah, I'm not worried about the diff process being unreliable, I'm worried about my memory being unreliable.  I'm worried that I'll be working on a hotfix for an old version of our product, so I'll have an old version of the database running, and I'll forget to wipe/reset the database when I go back to the current version.

> > (b) we're still experiencing some import failures related to timescaledb that make it hard for us to trust a diff based on an import of a running database.
> 
> Well, I've created pgModeler to be fully compatible with "original" PostgreSQL. So I can't guarantee that it'll work with other engines because I've never tested it with PostgreSQL forks like Timescaledb, Amazon RDS, Redshift, and etc... But nothing prevents us to patch the code to make pgModeler work with these databases. Just detail what's going wrong that I'll investigate.

FWIW, timescale isn't a fork - it's an extension.  It imports into standard postgres via `CREATE EXTENSION timescaledb;`.  I'll make a separate issue about the timescale import problems.

> Now, for the diff between two `.dbm` files, this can be even more complex than the current diff process. The diff between a model and a database is "less complicated" because the model generated from the import process does not contain dynamic objects like relationships that generate columns/constraints automatically, and some other objects that demand some other important validations. Anyway, the diff between models is completely feasible and I can plan to include it in future versions.

That's fair; I wasn't expecting this to be done _immediately_ or anything.  I wanted to document that it's a feature we would like - if and when you have time.  If the lack of this feature _really_ starts to get in our way, we can also submit a pull request - our team is all golang engineers, but we have some C/C++ experience.

**Comment by rkhaotix** [comment]
Hey @nelsam !

Sorry if my words sounded harsh. English is not my strong point! :)

Thanks for the clarification about timescaledb, I didn't know about it. For the feature request itself, I will think on something that covers your needs. I think we can start to work on it still in the development of 0.9.4.
